rust/lance/src/dataset/write/merge_insert.rs

```diff
@@ -2235,7 +2262,10 @@ mod tests {
             Field::new("id", DataType::UInt32, false),
             Field::new("value", DataType::UInt32, false),
         ]));
-        let concurrency = 10;
+        // To benchmark scaling curve: measure how long to run
+        //
+        // And vary `concurrency` to see how it scales. Compare this again `main`.
```

Comment:
The main benchmark I'd recommend running is just running this unit tests with the simulated object store latency. Run this for varying `concurrency` values in `2, 4, 8, 16, 32` on this branch and `main`. Goal is this should be as fast or faster than `main`. (Right now, it is slower.)